### PR TITLE
simplify code for ability used like weapon

### PR DESCRIPTION
**Pull request by newfrenchy83**
the use to special_active in attack_type::list_ability permit me to remove the static functions and to simplify the engine.

### CODE REVIEW

**Comment by newfrenchy83** [comment]
@gfgtdf , could you review this PR, please? This is not resolve of https://github.com/wesnoth/wesnoth/issues/4840 yet, but i hope find solution.

File: src/units/abilities.cpp [hunk 1]
```diff
@@ -1094,133 +1091,33 @@ namespace { // Helpers for attack_type::special_active()
...
-	if(other == nullptr) {
-		unit_map::const_iterator it = units.find(other_loc_);
-		if(it.valid()) {
-			other = it.get_shared_ptr().get();
-		}
-	}
-
-	// Make sure they're facing each other.
-	temporary_facing self_facing(self, self_loc_.get_relative_dir(other_loc_));
-	temporary_facing other_facing(other, other_loc_.get_relative_dir(self_loc_));
```
Comment (gfgtdf) [inline src/units/abilities.cpp:1214]: maybe you can explain a little more why this it no longer needed?

Comment (newfrenchy83) [reply src/units/abilities.cpp:1214]: @gfgtdf this code exist already in special_active, now what this function is used in list_leadership, it is unecessary to conservate a duplicate. see https://github.com/wesnoth/wesnoth/blob/49f44914b243faebfc327b26df7bae84e23f0683/src/units/abilities.cpp#L1185

**Comment by Pentarctagon** [comment]
Would it make sense to add some unit tests to help verify that everything still works as expected?

**Comment by newfrenchy83** [comment]
the test already exist in scenario-test  https://github.com/wesnoth/wesnoth/commit/ce4fb821737157761df0bab4d13e770a0efb23ec#diff-134e6f055615e6d49064a684872c530c

**Pull request merged**